svd-rs/src/addressblock.rs

```diff
@@ -60,3 +61,105 @@ impl AddressBlockUsage {
         }
     }
 }
+
+/// Builder for [`AddressBlock`]
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct AddressBlockBuilder {
+    offset: Option<u32>,
+    size: Option<u32>,
+    usage: Option<AddressBlockUsage>,
+    protection: Option<Protection>,
+}
+
+impl From<AddressBlock> for AddressBlockBuilder {
+    fn from(d: AddressBlock) -> Self {
+        Self {
+            offset: Some(d.offset),
+            size: Some(d.size),
+            usage: Some(d.usage),
+            protection: d.protection,
+        }
+    }
+}
+
+impl AddressBlockBuilder {
+    /// Set the offset of the block
+    pub fn offset(mut self, value: u32) -> Self {
+        self.offset = Some(value);
+        self
+    }
+    /// Set the size of the block
+    pub fn size(mut self, value: u32) -> Self {
+        self.size = Some(value);
+        self
+    }
+    /// Set the usage of the block
+    pub fn usage(mut self, value: AddressBlockUsage) -> Self {
+        self.usage = Some(value);
+        self
+    }
+    /// Set the protection of the block
+    pub fn protection(mut self, value: Option<Protection>) -> Self {
+        self.protection = value;
+        self
+    }
+    /// Validate and build a [`AddressBlock`].
+    pub fn build(self, lvl: ValidateLevel) -> Result<AddressBlock, SvdError> {
+        let mut de = AddressBlock {
+            offset: self
+                .offset
+                .ok_or_else(|| BuildError::Uninitialized("offset".to_string()))?,
+            size: self
+                .size
+                .ok_or_else(|| BuildError::Uninitialized("size".to_string()))?,
+            usage: self
+                .usage
+                .ok_or_else(|| BuildError::Uninitialized("usage".to_string()))?,
+            protection: self.protection,
+        };
+        if !lvl.is_disabled() {
+            de.validate(lvl)?;
+        }
+        Ok(de)
+    }
+}
+
+impl AddressBlock {
+    /// Make a builder for [`AddressBlock`]
+    pub fn builder() -> AddressBlockBuilder {
+        AddressBlockBuilder::default()
+    }
+    /// Modify an existing [`AddressBlock`] based on a [builder](AddressBlockBuilder).
+    pub fn modify_from(
+        &mut self,
+        builder: AddressBlockBuilder,
+        lvl: ValidateLevel,
+    ) -> Result<(), SvdError> {
+        if let Some(offset) = builder.offset {
+            self.offset = offset;
+        }
+        if let Some(size) = builder.size {
+            self.size = size;
+        }
+        if let Some(usage) = builder.usage {
+            self.usage = usage;
+        }
+        if builder.protection.is_some() {
+            self.protection = builder.protection;
+        }
+        if !lvl.is_disabled() {
+            self.validate(lvl)
+        } else {
+            Ok(())
+        }
+    }
+    /// Validate the [`AddressBlock`].
+    ///
+    /// # Notes
+    ///
+    /// This doesn't do anything.
+    pub fn validate(&mut self, _lvl: ValidateLevel) -> Result<(), SvdError> {
+        // TODO
```

Comment:
How is this a todo? What is there to do ?

Reply:
Don't know yet. Delete?

Reply:
Yeah I don't think there's a reason for this note 😅